dfastmi/io/RiversObject.py

```diff
@@ -86,54 +64,54 @@ def _read_rivers_file(self, filename):
         with open(filename, "r") as configfile:
             config.read_file(configfile)
 
-        self._river_data = RiverParameterData(config)
+        river_data = RiverParameterData(config)
 
         # initialize rivers dictionary
         iversion = self._validate_version_in_file(filename, config)
 
         self._verify_checksum_rivers(config, filename)
         
         # parse branches
+        self.branches = []
         self._parse_branches(config)
 
         # parse reaches and discharge locations
-        self._parse_reaches_and_discharge_locations(config)
+        self._parse_reaches(config, iversion)
 
         # collect the values for all other quantities
-        self.nreaches = [len(x) for x in self.allreaches]
+        #self.nreaches = [len(x) for x in self.allreaches]
 
         # initialize river data object
-        self._river_data.initialize(self.branches, self.nreaches)
+        #river_data.initialize(self.branches, self.nreaches)
         
         # call the specific reader for the file version
         if iversion == 1:
```

Comment:
at one point we should be able to do read_rivers and then leave it up to the class to fix if it is legacy or not